### DISCUSSION TITLE
Connection rest by peers and post example

### DISCUSSION THREAD

**Discussion by olup** [pschatzmann/arduino-audio-tools#1659] [Q&A] [answered]
Hllo, I am coming back to this library - great job for maintaining it year after year.

My problem is maybe not related to it, but you might have guidance on what's going on.

I am using a AI thinker audiokit 2.2 board that works with the AudioKitAC101 setup.
I tried a couple of example that worked great, but I am stumbling on the http POST example (https://github.com/pschatzmann/arduino-audio-tools/blob/d92a29dcea1e2d2320dfba431a8434cd7bc3d6da/examples/examples-audiokit/streams-audiokit-ram-audiokit/streams-audiokit-ram-audiokit.ino)

Everytime, a couple of packets are sent through (a variable number, maybe 4 or 5) and then the post fails with "connection reset by peers". I have a toy nodejs/express server on the other end and I can see the connection, but then no actual packets are received, then the connection ends.

I was wondering if you had a server iumplementation that you could share or maybe some guidance on some common error with esp32 network failures.

For reference, here is a log sample:

```
[I] SoundGenerator.h : 165 - SineWaveGenerator::begin(channels=2, sample_rate=44100, frequency=493.88)
[I] SoundGenerator.h : 149 - bool audio_tools::SineWaveGenerator<T>::begin() [with T = short int]
[I] AudioTypes.h : 128 - SoundGenerator: sample_rate: 44100 / channels: 2 / bits_per_sample: 16
[I] Buffers.h : 372 - resize: 4
[I] SoundGenerator.h : 192 - setFrequency: 493.88
[I] SoundGenerator.h : 193 - active: true
[I] AudioIO.h : 473 - byte range 0 - 10584001
[I] Url.h : 58 - Url::parse
[I] Url.h : 96 - url->http://192.168.1.140:9000/post
[I] Url.h : 97 - host->192.168.1.140
[I] Url.h : 98 - protocol->http
[I] Url.h : 99 - path->/post
[I] Url.h : 100 - port->9000
[I] HttpRequest.h : 220 - process connecting to host 192.168.1.140 port 9000
[I] HttpRequest.h : 340 - is connected true  with timeout 60000
[I] HttpRequest.h : 231 - Free heap: 240260
[I] HttpHeader.h : 276 - HttpHeader::write
[I] HttpHeader.h : 421 - -> POST /post HTTP/1.1
[I] HttpHeader.h : 210 -  -> Host: 192.168.1.140 
[I] HttpHeader.h : 210 -  -> Connection: keep-alive 
[I] HttpHeader.h : 210 -  -> Accept-Encoding: identity 
[I] HttpHeader.h : 210 -  -> Accept: */* 
[I] HttpHeader.h : 210 -  -> Content-Type: audio/pcm 
[I] HttpHeader.h : 342 -  -> <CR LF> 
[I] StreamCopy.h : 158 - StreamCopy::copy  1024 -> 1024 -> 1024 bytes - in 1 hops
[I] StreamCopy.h : 158 - StreamCopy::copy  1024 -> 1024 -> 1024 bytes - in 1 hops
[I] StreamCopy.h : 158 - StreamCopy::copy  1024 -> 1024 -> 1024 bytes - in 1 hops
[I] StreamCopy.h : 158 - StreamCopy::copy  1024 -> 1024 -> 1024 bytes - in 1 hops
[I] StreamCopy.h : 158 - StreamCopy::copy  1024 -> 1024 -> 1024 bytes - in 1 hops
[I] StreamCopy.h : 158 - StreamCopy::copy  1024 -> 1024 -> 1024 bytes - in 1 hops
[  4766][E][WiFiClient.cpp:429] write(): fail on fd 48, errno: 104, "Connection reset by peer"
[I] StreamCopy.h : 408 - try write  - 2 (open 1024 bytes) 
[I] StreamCopy.h : 408 - try write  - 3 (open 1024 bytes) 
...
```

**Comment by olup** [comment]
Actually the default sinc to post request is not suited to send chunked encoded streams. I build my own sink to post an audio stream and that is working as intended.

**Comment by pschatzmann** [comment, marked as answer]
That's correct: by default it is not chunked. 
If you want to do so, you would need to set the TRANSFER_ENCODING to CHUNKED in the request header.
